src/conn/conn_log.c

```diff
@@ -456,14 +455,14 @@ __wt_log_wrlsn(WT_SESSION_IMPL *session)
 	while (i < WT_SLOT_POOL) {
 		save_i = i;
 		slot = &log->slot_pool[i++];
+#ifdef	HAVE_DIAGNOSTIC
 		/*
-		 * XXX - During debugging I saw slot 0 become orphaned.
-		 * I believe it is fixed, but check for now.
-		 * This assertion should catch that.
+		 * Any slot in a zero state must be currently in progress.
 		 */
 		if (slot->slot_state == 0)
 			WT_ASSERT(session,
 			    slot->slot_release_lsn.file >= log->write_lsn.file);
+#endif
```

Comment:
Consider expressing this as:

```c
 			WT_ASSERT(session, slot->slot_state != 0 ||
 			    slot->slot_release_lsn.file >= log->write_lsn.file);
```

That avoids the `if` and any overhead during release builds without `#ifdef`.